### PR TITLE
Only try setting the execution permissions if the file doesn't have them already

### PR DESCRIPTION
**Pull request by azubieta**

### CODE REVIEW

File: src/libappimage/desktop_integration/integrator/Integrator.cpp [hunk 1]
```diff
@@ -270,7 +270,15 @@ namespace appimage {
                 }
 
                 void setExecutionPermission() {
-                    bf::permissions(appImage.getPath(), bf::owner_read | bf::owner_exe | bf::add_perms);
+                    if (access(appImage.getPath().c_str(), X_OK) != F_OK)
```
Comment (TheAssassin) [inline src/libappimage/desktop_integration/integrator/Integrator.cpp]: Please stick with boost:
https://stackoverflow.com/a/9776289

Comment (azubieta) [reply src/libappimage/desktop_integration/integrator/Integrator.cpp]: That was my first attempt, but it's not sufficient to check if the user has execution permissions, have to check if the user is the owner, also the group can have execution permissions then you have to check if the user belongs to the group and finally you can check if if the others have execution permissions.


`access` does all of that already and comes from `unistd.h` which is a header that we already depend on, therefore I think is better to reuse than to remake the wheel.

**Comment by azubieta** [comment]
I'm also thinking on a way on creating a uni-test for this scenario. Any ideas @probonopd

**Comment by probonopd** [comment]
Run the function on an AppImage that is already executable, and have the test fail if libappimage tries to set the executable bit again despite that it is already there?

**Comment by azubieta** [comment]
> fail if libappimage tries to set the executable

I have no way of intercept this without modifying the `libappimage` code.

**Comment by probonopd** [comment]
Why do you think the `libappimage` code should not be modified?

**Comment by azubieta** [comment]
> Why do you think the libappimage code should not be modified?

I guess that I didn't explain it properly. To the set execution permissions is performed by a single function call inside the register method. It would be required to extract that call and put it inside some kind of weird wrapper to be able to know from the tests environment whether it has been called or not. All of that will also reduce performance and will increase the complexity or our code base.

In this scenario the relation effort vs gain is really bad. That's why I don't want to do such modification on libappiimage.  

The ideal use case will be having an AppImage file without execution permission that doesn't belongs to the current user. We cold create such file but once we do `chown` on it we will lose control and this may just lead to weird things.

**Pull request merged**